src/PKSim.Core/Model/Formulation.cs

```diff
@@ -54,11 +75,60 @@ public virtual void AddRoute(string route)
       /// <summary>
       ///    returns true if the formulation can be used by the given route otherwise false
       /// </summary>
-      public virtual bool HasRoute(string route)
+      public virtual bool HasRoute(string route) => _allRoutes.Contains(route);
+
+      /// <summary>
+      /// Updates visibility and default state of parameters based on disperse system and particle size (only for Particle Dissolution formulation)
+      /// </summary>
+      public virtual void UpdateParticleParametersVisibility()
       {
-         return _allRoutes.Contains(route);
+         if (!IsParticleDissolution)
+            return;
+
+         //First ensure all parameters are visible. 
+         this.AllParameters().Each(showParticleParameter);
+
+         //Then hide parameters based on current selection
+         particleParametersToHide.Select(this.Parameter).Each(hideParticleParameter);
       }
 
+
+      private IEnumerable<string> particleParametersToHide
```

Comment:
The knowledge of which one to hide etc.. did not belong on presenter. This is probably better like that